packages/jellyfish-api-core/__tests__/category/poolpair/testpoolswap.test.ts

```diff
@@ -0,0 +1,183 @@
+    const dfiAddress = await getNewAddress(container)
+    const poolLiquidityAddress = await getNewAddress(container)
+
+    await createToken(container, 'CAT', { collateralAddress: tokenAddress })
+    await mintTokens(container, 'CAT', { address: dfiAddress })
+    await createPoolPair(container, 'CAT', 'DFI')
+    await addPoolLiquidity(container, {
+      tokenA: 'CAT',
+      amountA: 1000,
+      tokenB: 'DFI',
+      amountB: 500,
+      shareAddress: poolLiquidityAddress
+    })
+
+    const poolpairResult: PoolPairsResult = await container.call('getpoolpair', ['CAT-DFI'])
+    expect(Object.keys(poolpairResult).length).toStrictEqual(1)
+
+    // Note(canonbrother): simulate poolswap calculation to find reserveB
+    // case: swap 666 CAT to DFI
+    // 1000 : 500 = sqrt(1000 * 500) = 707.10678118
+    // 1666 : ? = sqrt(1666 * ?) = 707.10678118
+    // ? = 707.10678118^2 / 1666 = 300.120048014
+    // 1666 : 300.120048014 = 707.10678118
+    // 500 - 300.120048014 = 199.879951986  <-- testpoolswap returns
+    const poolpair: PoolPairInfo = Object.values(poolpairResult)[0]
+    const reserveAAfter: BigNumber = new BigNumber(poolpair.reserveA).plus(666)
+    const reserveBAfter: BigNumber = new BigNumber(poolpair.totalLiquidity).pow(2).div(reserveAAfter)
+
+    const result = await client.poolpair.testPoolSwap({
+      from: tokenAddress,
+      tokenFrom: 'CAT',
+      amountFrom: 666,
+      to: await getNewAddress(container),
+      tokenTo: 'DFI'
+    }) // 199.99999729@0
+
+    const testPoolSwapResultAmount = new BigNumber(result.split('@')[0]) // 199.99999729
+    const swapped = new BigNumber(poolpair.reserveB).minus(reserveBAfter) // 199.87995198635029880408
+
+    // Note(canonbrother): the result has slightly diff, use toFixed() to ignore the decimals
+    expect(swapped.toFixed(0)).toStrictEqual(testPoolSwapResultAmount.toFixed(0))
+  })
+
+  it('should testpoolswap with maxPrice', async () => {
+    expect.assertions(0)
+
+    const tokenAddress = await getNewAddress(container)
+    const dfiAddress = await getNewAddress(container)
+    const poolLiquidityAddress = await getNewAddress(container)
+
+    await createToken(container, 'ELF', { collateralAddress: tokenAddress })
+    await mintTokens(container, 'ELF', { address: dfiAddress })
+    await createPoolPair(container, 'ELF', 'DFI')
+    await addPoolLiquidity(container, {
+      tokenA: 'ELF',
+      amountA: 200,
+      tokenB: 'DFI',
+      amountB: 500,
+      shareAddress: poolLiquidityAddress
+    })
+
+    try {
+      await client.poolpair.testPoolSwap({
+        from: tokenAddress,
+        tokenFrom: 'ELF',
+        amountFrom: 666,
+        to: await getNewAddress(container),
+        tokenTo: 'DFI',
+        // reserveA/reserveB: 0.4
+        // reserveB/reserveA: 2.5
+        // if set maxPrice lower than 2.5 will hit error
+        maxPrice: 2.5
+      })
+    } catch (err) {
+      // should not reach here
+      expect(err).toBeUndefined()
+    }
```

Comment:
if you expect no error, you don't need to try to catch it? since if there is error it will fail the test